lib/kryptonite/aes.ex

```diff
@@ -143,25 +152,97 @@ defmodule Kryptonite.AES do
   end
 
   @doc """
-  Encrypt a stream using AES in CTR mode.
+  Encrypts a stream using AES in CTR mode.
 
   ## Examples
 
       iex> {key, iv} = {generate_key!(), Random.bytes!(16)}
-      iex> cypher = 'This is a secret' |> stream_encrypt(key, iv) |> Enum.to_list()
-      iex> is_list(cypher)
+      iex> 'This is a secret'
+      ...>     |> stream_encrypt(key, iv)
+      ...>     |> Enum.to_list()
+      ...>     |> is_list()
       true
   """
-  @spec stream_encrypt(Enumerable.t, key, iv) :: Enumerable.t
+  @spec stream_encrypt(Enumerable.t(), key, iv) :: Enumerable.t()
   def stream_encrypt(stream, key, iv) do
     acc0 = :crypto.stream_init(:aes_ctr, key, iv)
+
     reduce = fn elem, acc ->
       {acc, cypher} = :crypto.stream_encrypt(acc, elem |> List.wrap())
       {[cypher], acc}
     end
+
     Stream.transform(stream, acc0, reduce)
   end
 
+  @doc """ 
+  Encrypts + HMAC a stream into a Collectable
+
+  ## Examples
+
+      iex> {key, iv} = {generate_key!(), Random.bytes!(16)}
+      iex> File.write!("/tmp/plain.txt", "This is a secret")
```

Comment:
Race condition here when running parallel builds. I'd probably add a timestamp based identifier to the filename. WDYT?